File: 2779-maximum-beauty-of-an-array-after-applying-operation/2779-maximum-beauty-of-an-array-after-applying-operation.java
class Solution {
    public int maximumBeauty(int[] nums, int k) {
        int val = 100000;
        int[] fre = new int[1000000];
        for(int num : nums) {
            fre[num-k+val]++;
            fre[num+k+val+1]--;
        }
        int sum=0;
        int ans=0;
        for(int i=0;i<fre.length;i++) {
            sum+=fre[i];
            ans=Math.max(ans,sum);
        }
        return ans;
    }
}

File: 2779-maximum-beauty-of-an-array-after-applying-operation/2779-maximum-beauty-of-an-array-after-applying-operation.cpp
class Solution {
public:
    int maximumBeauty(vector<int>& nums, int k) {
        vector<int> fre(300005);
        for(auto x: nums){
            int ri=x+k+100000,le=x+100000-k;
            fre[le]+=1;
            fre[ri+1]-=1;
        }
        for(int i=1;i<fre.size();i++)
            fre[i]+=fre[i-1];
        int maxi=0;
        for(auto x: fre)
            maxi=max(maxi,x);
        return maxi;
    }
};